src/main/java/co/cask/hydrator/plugin/batchSource/KafkaBatchSource.java

```diff
@@ -319,17 +317,17 @@ public void validate() {
       // if format is empty, there must be just a single message field of type bytes or nullable types.
       if (Strings.isNullOrEmpty(format)) {
         List<Schema.Field> messageFields = messageSchema.getFields();
-        if (messageFields.size() > 1) {
-          List<String> fieldNames = new ArrayList<>();
-          for (Schema.Field messageField : messageFields) {
-            fieldNames.add(messageField.getName());
-          }
-          throw new IllegalArgumentException(String.format(
-            "Without a format, the schema must contain just a single message field of type bytes or nullable bytes. " +
-              "Found %s message fields (%s).", messageFields.size(), Joiner.on(',').join(fieldNames)));
-        }
-
-        Schema.Field messageField = messageFields.get(0);
+//        if (messageFields.size() > 1) {
```

Comment:
The messageFields size will be greater than 1 when we add the key, offset, and time fields to the output schema, which will always cause the exception to be thrown.